core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
+      String realm = getConfiguration().getString(RestConfig.AUTHENTICATION_REALM);
+      List<String> roles = getConfiguration().getList(RestConfig.AUTHENTICATION_ROLES);
+      final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
+      Constraint constraint = new Constraint();
+      constraint.setAuthenticate(true);
+      constraint.setRoles(roles.toArray(new String[roles.size()]));
+      ConstraintMapping constraintMapping = new ConstraintMapping();
+      constraintMapping.setConstraint(constraint);
+      constraintMapping.setMethod("*");
+      constraintMapping.setPathSpec("/*");
+      securityHandler.addConstraintMapping(constraintMapping);
```

Comment:
To follow up on offline discussion, I think this works fine -- use this to provide auth to the entire set of resources, but then have each resource manage authentication on its own. I think we want to remove the `AUTHENTICATION_ROLES_CONFIG` instead allowing this to be passed into the relevant methods (which I guess need to be made non-static). This allows the applications using these classes to figure out how to compute the union of all roles they want to allow into any resource, pass that in, then within each handler deal with adding the `SecurityContext` and filtering by the role appropriate for the handler on their own. This also allows them to figure out how best to express which resources/methods should be allowed for each role, e.g. adding higher level classifications rather than expressing them on a per-endpoint/method basis.